### ISSUE TITLE
Shibboleth users who predate Shibboleth are assumed to be email-verified but aren't

### ISSUE THREAD

**Issue by donsizemore** [IQSS/dataverse#5663]
This issue is related to #1401 and possibly supersedes #3300. Per Phil and Pete in [IRC](http://irclog.iq.harvard.edu/dataverse/2019-03-18#i_88437) Shibboleth users are now assumed to have their e-mail addresses verified as trusted / provided by the IdP. For new users this happens. Existing users such as myself and our archivists remain unverified. I can sign in with my Shibboleth account to our test server running 4.11 and don't get an "emailconfirmed" timestamp:

```
dvntest=> select emailconfirmed from authenticateduser where email='dls@email.unc.edu';
 emailconfirmed 
----------------

(1 row)
```

I could start wedging in timestamps for institutional users but would love to find the logic snag.

**Comment by pameyer** [comment]
I _think_ the logic snag is that when the `authenticateduser.emailconfirmed` column was added in 4.5.1, it wasn't accompanied by an update setting that column to account creation time for shibboleth users.

**Comment by donsizemore** [comment]
@pameyer seems better to make current Shib logins check for emailconfirmed and set if necessary — we have a bunch of users matching unc.edu who may or may not sign in going forward. My inner security Chicken Little would prefer to leave them NULL.

**Comment by pameyer** [comment]
@donsizemore That seems reasonable to me; but means it would take code changes rather than database cleanup.

**Comment by pdurbin** [comment]
@donsizemore great. Update at each login. Sounds like a plan.

OAuth and OIDC users would not be affected by any change I'm planning for this issue. For OIDC, please note that there's an open issue about email verification:

- #6679

Meanwhile I did some more Shib testing locally and wanted to share some screenshots and a bug I'd like to fix.

When you create a Shib account, the process is fine (email shows as verified) except that the in-app notification says, "check for your welcome email to verify your address." This part of the message shouldn't be shown to Shib users. The email itself is fine. That is, the email correctly does not include a link or any text about verifying your email.

![Screen Shot 2022-04-06 at 12 24 27 PM](https://user-images.githubusercontent.com/21006/162056139-5a2d327d-e20b-43c0-9218-6e9058f3f599.png)
![Screen Shot 2022-04-06 at 12 27 41 PM](https://user-images.githubusercontent.com/21006/162056154-9710ae31-9272-4de6-ad3f-86dfe9243dc1.png)
![Screen Shot 2022-04-06 at 12 27 55 PM](https://user-images.githubusercontent.com/21006/162056160-d1ce6eb1-4f93-415f-a991-6f7519a110f0.png)

---

Subject: Root: Your account has been created

Hello, 
Welcome to Root! Get started by adding or finding data. Have questions? Check out the User Guide at https://guides.dataverse.org/en/5.10/user or contact Root Support Team at support@mailinator.com for assistance.

You may contact us for support at support@mailinator.com.

Thank you,
Root Support Team

---

I have the change queued up locally to update the in-app notification for newly created Shib users.

**Comment by pdurbin** [comment]
More screenshots. I also looked at converting accounts from builtin to Shib. The bug here is that the email is not automatically set to verified.

![Screen Shot 2022-04-06 at 2 09 30 PM](https://user-images.githubusercontent.com/21006/162058101-101b5e26-4b83-4fa4-9de4-848a99832a66.png)
![Screen Shot 2022-04-06 at 2 10 08 PM](https://user-images.githubusercontent.com/21006/162058104-e7f60ab4-eb8c-4d57-9600-dd8625000bdf.png)

Another potential bug is that a "confirm email" token is in the database for this user. Presumably this token was created back when I created the builtin user. Perhaps the conversion process to Shib should clear out any "confirm email" tokens. In practice, this is not a big problem because this token will simply expire in 24 hours (by default).
